tsl/src/nodes/decompress_chunk/pred_text.c

```diff
@@ -0,0 +1,159 @@
+							 uint64 *restrict result)
+{
+	Assert(!arrow->dictionary);
+
+	text *consttext = (text *) DatumGetPointer(constdatum);
+	const size_t textlen = VARSIZE_ANY_EXHDR(consttext);
+	const uint8 *cstring = (uint8 *) VARDATA_ANY(consttext);
+	const uint32 *offsets = (uint32 *) arrow->buffers[1];
+	const uint8 *values = (uint8 *) arrow->buffers[2];
+
+	const size_t n = arrow->length;
+	for (size_t outer = 0; outer < n / 64; outer++)
+	{
+		uint64 word = 0;
+		for (size_t inner = 0; inner < 64; inner++)
+		{
+			const size_t row = outer * 64 + inner;
+			const size_t bit_index = inner;
+#define INNER_LOOP                                                                                 \
+	const uint32 start = offsets[row];                                                             \
+	const uint32 end = offsets[row + 1];                                                           \
+	Assert(end >= start);                                                                          \
+	const uint32 veclen = end - start;                                                             \
+	bool isequal = veclen != textlen ?                                                             \
+					   false :                                                                     \
+					   (strncmp((char *) &values[start], (char *) cstring, textlen) == 0);         \
+	word |= ((uint64) (isequal == needequal)) << bit_index;
+
+			INNER_LOOP
+		}
+		result[outer] &= word;
+	}
+
+	if (n % 64)
+	{
+		uint64 word = 0;
+		for (size_t row = (n / 64) * 64; row < n; row++)
+		{
+			const size_t bit_index = row % 64;
+			INNER_LOOP
+		}
+		result[n / 64] &= word;
+	}
+
+#undef INNER_LOOP
+}
+
+void
+vector_const_texteq(const ArrowArray *arrow, const Datum constdatum, uint64 *restrict result)
+{
+	vector_const_text_comparison(arrow, constdatum, /* needequal = */ true, result);
+}
+
+void
+vector_const_textne(const ArrowArray *arrow, const Datum constdatum, uint64 *restrict result)
+{
+	vector_const_text_comparison(arrow, constdatum, /* needequal = */ false, result);
+}
+
+/*
+ * Generate specializations for LIKE functions based on database encoding. This
+ * follows the Postgres code from backend/utils/adt/like.c.
+ */
+
+#define LIKE_TRUE 1
+#define LIKE_FALSE 0
+#define LIKE_ABORT (-1)
+
+/* setup to compile like_match.c for UTF8 encoding, using fast NextChar */
+#define NextByte(p, plen) ((p)++, (plen)--)
+#define NextChar(p, plen)                                                                          \
+	do                                                                                             \
+	{                                                                                              \
+		(p)++;                                                                                     \
+		(plen)--;                                                                                  \
+	} while ((plen) > 0 && (*(p) &0xC0) == 0x80)
```

Comment:
Could you add a comment on why the `*(p) &0xC0) == 0x80` part is needed and what the `0x80` constant represents?

Reply:
This follows the PG code as well, I'll add more details to the comment above.

Reply:
I think it's just skipping the UTF8-encoded characters, and this check is checking for the UTF8 prefix. I try to avoid adding comments to the copied PG code, because it makes comparing with the original more difficult.

Reply:
That is true. I compared it with the PG code and my function looked a bit different. So, I was not sure if you modified it or if this is upstream code from another version. If we have a comment about the PG version of the function, this should be fine.